### PR TITLE
Work around deprecated/removed DatabaseBase::getSearchEngine

### PR DESCRIPTION
**Pull request by s7eph4n**
This PR is made in reference to: https://phabricator.wikimedia.org/T147683

This PR addresses or contains:
DatabaseBase::getSearchEngine was removed in 1.28.

This PR includes:
- [x] Tests (unit/integration): Existing tests failed. No new tests necessary.
- [ ] CI build passed: No, but not because of this patch.

### CODE REVIEW

**Comment by kghbln** [comment]
There is also a [unit test](https://github.com/SemanticMediaWiki/SemanticMediaWiki/blob/c87e501aa908d55481d0538179d712a0695dbb44/tests/phpunit/Unit/MediaWiki/Search/SearchTest.php) involving this. This probably needs to be  adjusted too.

No backport to 2.4 required since this one is not meant to support MW 1.28+

**Comment by s7eph4n** [comment]
No need to fix the unit test. The existing test fails without the patch, works with the patch.

**Comment by kghbln** [comment]
Great! Everything is fluffy I guess. :)

**Comment by JeroenDeDauw** [comment]
Yay MediaWiki, the new thing does not have dependency injection either. At least it's better named crap code...

**Comment by JeroenDeDauw** [comment]
Thanks!

**Comment by JeroenDeDauw** [comment]
@s7eph4n  Note that there is a follow up at https://github.com/SemanticMediaWiki/SemanticMediaWiki/pull/1898 which highlights some things you could have done differently

**Comment by mwjames** [comment]
That was a bit quick here because of:

> No need to fix the unit test. The existing test fails without the patch, works with the patch.

https://travis-ci.org/SemanticMediaWiki/SemanticMediaWiki/jobs/165969263

Not really, now it returns with:

```
3) SMW\Tests\MediaWiki\Search\SearchTest::testGetDefaultFallbackSearchEngineForNullFallbackSearchType
Failed asserting that SearchEngineDummy Object (...) is an instance of class "SearchDatabase".
```

**Comment by mwjames** [comment]
> There is also a unit test involving this. This probably needs to be adjusted too.

https://travis-ci.org/SemanticMediaWiki/SemanticMediaWiki/jobs/165436085

The unit test is fine since we mock that part which isn't really helpful (no failure in above master test from two days ago) but if you are interested in a real integration test.

If I have time, I'll add an integration test so things like below will shine through.

````
Fatal error: Call to undefined method DatabaseMysqli::getSearchEngine() in ...\extensions\SemanticMediaWiki\src\MediaWiki\Search\Search.php on line 58
```

PS: There is another case https://phabricator.wikimedia.org/T147550 where WMF doesn't mind its own deprecation policy but luckily our integration test shouted out on this one.

PSS: 1.28/master is allowed to fail otherwise we go crazy on the extend exposition of WMF funnyism.

**Comment by JeroenDeDauw** [comment]
> PSS: 1.28/master is allowed to fail

Muh... I want to be able to merge PRs if the CI is green

**Comment by mwjames** [comment]
> I want to be able to merge PRs if the CI is green

I want to no have to FIX (or circumvent) issues not related to SMW-core but surely that's not going to happen.

**Comment by mwjames** [comment]
#1825 is a prime example of what happens when you run master.

File: src/MediaWiki/Search/Search.php [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 use Content;
 use DatabaseBase;
+use MediaWiki\MediaWikiServices;
```
Comment (mwjames) [inline src/MediaWiki/Search/Search.php]: Access, reference, and usage to `MediaWiki\MediaWikiServices` should be confined to either `SMW\Tests\TestEnvrionment` [0] or `ApplicationFactory`.

[0] https://github.com/SemanticMediaWiki/SemanticMediaWiki/blob/master/tests/phpunit/TestEnvironment.php#L95-L114

**Comment by mwjames** [comment]
> If I have time, I'll add an integration test so things like below will shine through.

... And it shall be light #1904.

**Pull request merged**